lib/Jobs/AutomaticLocationMapperJob.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+declare(strict_types=1);
+/**
+ * @copyright Copyright (c) 2022 Louis Chemineau <louis@chmn.me>
+ *
+ * @author Louis Chemineau <louis@chmn.me>
+ *
+ * @license AGPL-3.0-or-later
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Photos\Jobs;
+
+use OCA\Photos\AppInfo\Application;
+use OCA\Photos\Service\MediaLocationManager;
+use OCP\AppFramework\Utility\ITimeFactory;
+use OCP\BackgroundJob\TimedJob;
+use OCP\Files\Folder;
+use OCP\Files\IRootFolder;
+use OCP\IConfig;
+use OCP\IUserManager;
+
+class AutomaticLocationMapperJob extends TimedJob {
+	public function __construct(
+		ITimeFactory $time,
+		private IConfig $config,
+		private IRootFolder $rootFolder,
+		private IUserManager $userManager,
+		private MediaLocationManager $mediaLocationManager,
+	) {
+		parent::__construct($time);
+		$this->mediaLocationManager = $mediaLocationManager;
+
+		$this->setTimeSensitivity(\OCP\BackgroundJob\IJob::TIME_INSENSITIVE);
+		$this->setInterval(24 * 3600);
+	}
+
+	protected function run($argument) {
+		$lastUser = $this->config->getAppValue(Application::APP_ID, 'lastMappedUser', '');
+		$startTime = null;
+		$users = $this->userManager->search('');
+
+		if ($lastUser === '') {
+			$lastUser = $users[array_key_first($users)]->getUID();
+		}
+
+		foreach ($users as $user) {
+			if ($startTime === null) {
+				// Skip all user before lastUser.
+				if ($lastUser === $user->getUID()) {
+					$startTime = time();
+				}
+
+				continue;
+			}
+
+			// Stop if execution time is more than one hour.
+			if (time() - $startTime > 60 * 60) {
+				return;
+			}
+
+			$this->scanFilesForUser($user->getUID());
+			$this->config->setAppValue(Application::APP_ID, 'lastMappedUser', $user->getUID());
```

Comment:
@ArtificialOwl any way to properly kill this job ? Currently, after having loop over every user, it will still get called regularly, but won't do anything.